www/api/class/centreon_metric.class.php

```diff
@@ -125,8 +148,14 @@ protected function getListByService()
             AND s.service_id = i.service_id
             AND h.enabled = 1
             AND s.enabled = 1
-            AND CONCAT(h.name," - ", s.description, " - ",  m.metric_name) LIKE :name
-            ORDER BY CONCAT(h.name," - ", s.description, " - ",  m.metric_name) COLLATE utf8_general_ci ';
+            AND CONCAT(h.name," - ", s.description, " - ",  m.metric_name) LIKE :name ';
+
+        if (!$isAdmin) {
+            $acl = new CentreonACL($userId, $isAdmin);
+            $aclMetrics .= 'AND s.service_id IN (' . $acl->getServicesString('ID', $this->pearDBMonitoring) . ') ';
+        }
+
+        $query .= ' ORDER BY CONCAT(h.name," - ", s.description, " - ",  m.metric_name) COLLATE utf8_general_ci ';
```

Comment:
could you try with the alias defined at line 144 ?
```suggestion
        $query .= ' ORDER BY fullname COLLATE utf8_general_ci ';
```